doc/release/upcoming_changes/15900.deprecation.rst

```diff
@@ -0,0 +1,16 @@
+Indexing errors will be reported even when index result is empty
+----------------------------------------------------------------
+In the future, NumPy will raise an index error when an
+integer array index contains out of bound values even if a non-indexed
+dimension is of length 0.
+This can happen when the array is previously empty, or an empty
+slice is involved::
```

Comment:
I might be good to give an example of each of these two cases.

The other example could be `arr2 = arr[:, :0]`, `arr2[[20]]`

Reply:
I applied those doc suggestions, and added an `np.zeros((5, 0))` example